### PR TITLE
Fixes intellisense for vscode and VS, it also enables testing in Test Explorer

### LINKED ISSUE #34657: Visual Studio Test Explorer is broken because the OutputPath isn't updated correctly in the Design Time Build
Repro: `.\build.cmd -vs System.Text.RegularExpressions`

Run a test in the Test Explorer and see the errors in the Test Explorer Output Window:
```
[4/7/2020 11:05:23.856 PM] ---------- Starting test run ----------
[4/7/2020 11:05:23.875 PM] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection 
[4/7/2020 11:05:23.875 PM] Event:VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection key: VS.UnitTest.TestWindow.RunSettingsProvider.UserFile value:False
[4/7/2020 11:05:23.875 PM] Event:VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection key: VS.UnitTest.TestWindow.RunSettingsProvider.SolutionFile value:False
[4/7/2020 11:05:23.875 PM] Event:VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection key: VS.UnitTest.TestWindow.RunSettingsProvider.AutoDetect value:False
[4/7/2020 11:05:24.059 PM] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettingsService 
[4/7/2020 11:05:24.059 PM] Event:VS/UnitTest/TestWindow/Ext/RunSettingsService key: VS.UnitTest.TestWindow.RunSettingsService.Name value:VSTest Run Configuration
[4/7/2020 11:05:24.060 PM] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettingsService 
[4/7/2020 11:05:24.060 PM] Event:VS/UnitTest/TestWindow/Ext/RunSettingsService key: VS.UnitTest.TestWindow.RunSettingsService.Name value:Fakes
[4/7/2020 11:05:24.076 PM] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettingsService 
[4/7/2020 11:05:24.076 PM] Event:VS/UnitTest/TestWindow/Ext/RunSettingsService key: VS.UnitTest.TestWindow.RunSettingsService.Name value:Dynamic Code Coverage
[4/7/2020 11:05:24.077 PM] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettings 
[4/7/2020 11:05:24.077 PM] Event:VS/UnitTest/TestWindow/Ext/RunSettings key: VS.UnitTest.TestWindow.RunSettings.Services value:3
[4/7/2020 11:05:24.079 PM] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection 
[4/7/2020 11:05:24.079 PM] Event:VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection key: VS.UnitTest.TestWindow.RunSettingsProvider.ProjectFileCount value:0
[4/7/2020 11:05:24.079 PM] Event:VS/UnitTest/TestWindow/RunSettingsProvider/FileSelection key: VS.UnitTest.TestWindow.RunSettingsProvider.ContainerCount value:1
[4/7/2020 11:05:24.130 PM] Some tests from the test run selection will be executed by name.
[4/7/2020 11:05:24.265 PM] Tests run settings for C:\git\runtime3\artifacts\bin\System.Text.RegularExpressions.Tests\Debug\System.Text.RegularExpressions.Tests.dll:
 <RunSettings>
  <RunConfiguration>
    <ResultsDirectory>C:\git\runtime3\src\libraries\System.Text.RegularExpressions\TestResults</ResultsDirectory>
    <SolutionDirectory>C:\git\runtime3\src\libraries\System.Text.RegularExpressions\</SolutionDirectory>
    <CollectSourceInformation>False</CollectSourceInformation>
  </RunConfiguration>
</RunSettings>.
[4/7/2020 11:05:24.555 PM] Trying to update view -962344839
[4/7/2020 11:05:24.555 PM] Updating view -962344839
[4/7/2020 11:05:24.558 PM] Updating view -962344839
[4/7/2020 11:05:24.576 PM] UpdateSummary Detail Unchanged: SKIPPED
[4/7/2020 11:05:24.834 PM] UpdateSummary Detail Unchanged: SKIPPED
[4/7/2020 11:05:24.938 PM] Logging TestHost Diagnostics in file: C:\Users\vihofer\AppData\Local\Temp\TestPlatformLogs\15428_04_07_2020_23_04_18\logs.host.20-04-07_23-05-24_50967_6.txt
[4/7/2020 11:05:25.986 PM] Could not find file C:\git\runtime3\artifacts\bin\System.Text.RegularExpressions.Tests\Debug\System.Text.RegularExpressions.Tests.dll.
[4/7/2020 11:05:25.987 PM] None of the specified source(s) 'C:\git\runtime3\artifacts\bin\System.Text.RegularExpressions.Tests\Debug\System.Text.RegularExpressions.Tests.dll' is valid. Fix the above errors/warnings and then try again. 
[4/7/2020 11:05:26.371 PM] ========== Test run finished: 0 Tests run in 2.5 sec (0 Passed, 0 Failed, 0 Skipped) ==========
```

Additional logs indicate that the OutputPath is read incorrectly:

```
===
Id: 2c58640b-5bed-4e83-9554-cd2b9762643f
ProjectFilePath: C:\git\runtime3\src\libraries\System.Text.RegularExpressions\src\System.Text.RegularExpressions.csproj
DefaultOutputPath: C:\git\runtime3\artifacts\bin\System.Text.RegularExpressions\Debug\System.Text.RegularExpressions.dll
DefaultTargetFramework: net5.0
ProjectName: System.Text.RegularExpressions
Capabilities: UseFileGlobs,AppDesigner,AssemblyReferences,Managed,SupportAvailableItemName,FolderPublish,CPS,ProjectConfigurationsDeclaredDimensions,RelativePathDerivedDefaultNamespace,OpenProjectFile,UserSourceItems,VisualStudioWellKnownOutputGroups,DynamicDependentFile,NoGeneralDependentFileIcon,CSharp,IntegratedConsoleDebugging,SharedProjectReferences,Publish,EditAndContinue,Microsoft.VisualStudio.ProjectSystem.RetailRuntime,AllTargetOutputGroups,SingleFileGenerators,ProjectReferences,HostSetActiveProjectConfiguration,ClassDesigner,PackageReferences,GenerateDocumentationFile,HandlesOwnReload,RunningInVisualStudio,WinRTReferences,Pack,LaunchProfiles,SharedImports,LanguageService,PersistDesignTimeDataOutOfProject,DependenciesTree,DeclaredSourceItems,PreserveFormatting,.NET,DataSourceWindow,OutputGroups,COMReferences,ProjectImportsTree
IsAppContainer: False
IsCpsProject: True
IsMiscellaneous: False
HasTestContainerCapability: False
IsCSharpOrVBProject: True
```

This likely broke with https://github.com/dotnet/runtime/commit/9a14b86ae54ef726819a7be34dac114583d83189.

Removing the `DesignTimeBuild` condition in https://github.com/dotnet/runtime/blob/c4507d795038c41fc041276e0737045262f9980d/src/libraries/Directory.Build.props#L173 fixes the issue as the OutputPath is set correctly.

cc @ericstj @Anipik @safern

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/34657

### CODE REVIEW

File: src/libraries/intellisense.targets [hunk 1]
```diff
@@ -17,7 +17,11 @@
     <TargetFramework>$(TargetFramework.SubString(0, $(TargetFramework.IndexOf('-'))))</TargetFramework>
   </PropertyGroup>
 
-    <!-- Split $(TargetFramework) (e.g. net45) into short identifier and short version (e.g. 'net' and '45'). -->
+  <PropertyGroup>
+    <RefPath>$([MSBuild]::NormalizeDirectory('$(RefRootPath)', '$(TargetFramework)'))</RefPath>
```
Comment (safern) [inline src/libraries/intellisense.targets:21]: Why not instead set `BuildTargetFrameworkRefPath`? It seems like `RefPath` takes its value:
https://github.com/dotnet/runtime/blob/952eecabbdd57540f4fe72c490c794e79c3d0cf3/src/libraries/Directory.Build.props#L257

File: src/libraries/intellisense.targets [hunk 2]
```diff
@@ -75,6 +79,7 @@
     <ProjectAssetsCacheFile Condition="'$(ProjectAssetsCacheFile)' == ''">$(IntermediateOutputPath)$(MSBuildProjectName).assets.cache</ProjectAssetsCacheFile>
     <ProjectAssetsCacheFile>$([MSBuild]::NormalizePath($(MSBuildProjectDirectory), $(ProjectAssetsCacheFile)))</ProjectAssetsCacheFile>
     <NuGetTargetMoniker>$(TargetFrameworkMoniker)</NuGetTargetMoniker>
+    <TargetPath>$(OutputPath)$(TargetFileName)</TargetPath>
```
Comment (safern) [inline src/libraries/intellisense.targets:82]: Why is this needed?

Comment (Anipik) [reply src/libraries/intellisense.targets:82]: TargetPath is the final path to the test dll, it is calculated using the output path in .targets file. After we override the output path we need to update the targetPath as well.

File: src/libraries/intellisense.targets (hunk 1 above)
Comment (Anipik) [reply src/libraries/intellisense.targets:21]: When we are in a VS, and we want to run tests or build stuff for net472 or some other framework than net50, there is no way to override the default value of BuildTargetFramework.
Refpaths for all the designTimeBuilds will also have the netcoreapp50 refpath only.

Comment (safern) [reply src/libraries/intellisense.targets:21]: I mean setting `BuildTargetFrameworkRefPath` here instead of `RefPath`. Does Directory.Build.targets come into play in DesignTime build?

File: src/libraries/intellisense.targets (hunk 2 above)
Comment (ericstj) [reply src/libraries/intellisense.targets:82]: You need to add comments here describing why you do this sort of thing.

File: src/libraries/intellisense.targets (hunk 1 above)
Comment (ericstj) [reply src/libraries/intellisense.targets:21]: @safern BuildTargetFrameworkRefPath is the value for the vertical, not the current configuration.  BuildTargetFrameworkRefPath doesn’t need to be reset: it’s value remains the same and is only set by root level build invocation. RefPath on the other hand is dependent on project TargetFramework and needs to be reset here since.

**Review by ericstj — APPROVED** [review]
Please add comments explaining

File: src/libraries/intellisense.targets (hunk 1 above)
Comment (safern) [reply src/libraries/intellisense.targets:21]: I see, I missed the condition 😄

**Review by safern — APPROVED** [review]
Modulo adding the comment.
